mod-source-record-manager-server/src/main/java/org/folio/rest/impl/MappingRulesProviderImpl.java

```diff
@@ -33,11 +34,13 @@ public MappingRulesProviderImpl(Vertx vertx, String tenantId) { //NOSONAR
 
   @Override
   public void getMappingRulesByRecordType(String recordType, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    Record.RecordType enumRecordType = QueryPathUtil.toRecordType(recordType)
+      .orElseThrow(() -> new BadRequestException("Only marc-bib or marc-holdings supports"));
```

Comment:
Why did you move it out from future flow? With your change where `BadRequestException` will be handled?